### PR TITLE
Add unit tests for UrdfJoints

### PR DESCRIPTION
**Pull request by peifeng-unity**
## Proposed change(s)

Add unit tests to the subclasses of urdf joint. The UrdfJoint part may need to be refactored; added [AIRO-688](https://jira.unity3d.com/browse/AIRO-688) to follow up.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Documentation update
- [x] Other: add unit tests

## Testing and Verification

Yamato jobs in this PR

## Checklist
- [x] Ensured this PR is up-to-date with the `dev` branch
- [x] Created this PR to target the `dev` branch
- [ ] Followed the style guidelines as described in the [Contribution Guidelines](../CONTRIBUTING.md)
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the documentation as appropriate

## Other comments

### CODE REVIEW

File: com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuousTests.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    public void SetAxisOfMotion(Vector3 axisofMotion) => this.axisofMotion = axisofMotion;
+    public void Dynamics(Joint.Dynamics dynamics) => SetDynamics(dynamics);
+}
+
+public class UrdfJointContinuousTests
+{
+    [Test]
+    public void Create_Succeeds()
```
Comment (at669) [inline com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuousTests.cs:27]: general question: does it better follow convention if this were `Create_ContinuousJoint_Succeeds()` or would calling out specifically `Continuous` be unnecessary given that these are all Continuous tests?

File: com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuousTests.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        articulationBody.jointPosition = new ArticulationReducedSpace(1, 2, 3);
+        articulationBody.jointVelocity = new ArticulationReducedSpace(4, 5, 6);
+        articulationBody.jointForce = new ArticulationReducedSpace(7, 8, 9);
+
+        Assert.AreEqual(1, joint.GetPosition());
+        Assert.AreEqual(4, joint.GetVelocity());
+        Assert.AreEqual(7, joint.GetEffort());
+
+        Object.DestroyImmediate(baseObject);
+        Object.DestroyImmediate(linkObject);
```
Comment (at669) [inline com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuousTests.cs:60]: question: I always thought destroying a parent would automatically destroy the children--would this cause any issues? and/or am I mistaken and should update my unit tests to also destroy children individually?

File: com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuousTests.cs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        linkObject.transform.parent = baseObject.transform;
+
+        UrdfJoint.Create(baseObject, UrdfJoint.JointTypes.Fixed);
+        UrdfJoint joint = UrdfJointContinuous.Create(linkObject);
+        ArticulationBody articulationBody = linkObject.GetComponent<ArticulationBody>();
+
+        Assert.AreEqual(0, articulationBody.xDrive.target);
+        joint.UpdateJointState(1);
+        Assert.AreEqual(1, articulationBody.xDrive.target);
+    }
```
Comment (at669) [inline com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuousTests.cs:77]: destroy object

File: com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismaticTests.cs [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        GameObject linkObject = new GameObject("link");
+        linkObject.transform.parent = baseObject.transform;
+
+        UrdfJoint.Create(baseObject, UrdfJoint.JointTypes.Fixed);
+        UrdfJoint joint = UrdfJointPrismatic.Create(linkObject);
+        ArticulationBody articulationBody = linkObject.GetComponent<ArticulationBody>();
+
+        Assert.AreEqual(0, articulationBody.xDrive.target);
+        joint.UpdateJointState(1);
+        Assert.AreEqual(1, articulationBody.xDrive.target);
```
Comment (at669) [inline com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismaticTests.cs:79]: destroy object

File: com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointRevoluteTests.cs [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        GameObject linkObject = new GameObject("link");
+        linkObject.transform.parent = baseObject.transform;
+
+        UrdfJoint.Create(baseObject, UrdfJoint.JointTypes.Fixed);
+        UrdfJoint joint = UrdfJointRevolute.Create(linkObject);
+        ArticulationBody articulationBody = linkObject.GetComponent<ArticulationBody>();
+
+        Assert.AreEqual(0, articulationBody.xDrive.target);
+        joint.UpdateJointState(1);
+        Assert.AreEqual(1 * Mathf.Rad2Deg, articulationBody.xDrive.target);
```
Comment (at669) [inline com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointRevoluteTests.cs:79]: destroy object

File: com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuousTests.cs (hunk 1 above)
Comment (peifeng-unity) [reply com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuousTests.cs:27]: Good point. I am not exactly sure but feel that `Create_ContinuousJoint_Succeeds` might be better. Will update it.

File: com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuousTests.cs (hunk 2 above)
Comment (peifeng-unity) [reply com.unity.robotics.urdf-importer/Tests/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuousTests.cs:60]: no, you are right that the destroy method will remove the children. I didn't realize it. Will update. Thanks!

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPlanar.cs [hunk 6]
```diff
@@ -178,11 +181,18 @@ protected override void AdjustMovement(Joint joint)
             {
                 unityJoint.linearLockY = ArticulationDofLock.LimitedMotion;
                 unityJoint.linearLockZ = ArticulationDofLock.LimitedMotion;
-                ArticulationDrive drive = unityJoint.xDrive;
-                drive.upperLimit = (float)joint.limit.upper;
-                drive.lowerLimit = (float)joint.limit.lower;
+                var drive = new ArticulationDrive()
```
Comment (sdiao) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPlanar.cs:184]: refactor looks good.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs [hunk 7]
```diff
@@ -74,7 +73,7 @@ public override float GetPosition()
         public override float GetVelocity()
         {
 #if UNITY_2020_1_OR_NEWER
-            return unityJoint.velocity[xAxis];
+            return unityJoint.jointVelocity[xAxis];
```
Comment (sdiao) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs:76]: curious is velocity and jointVelocity the same concept?

**Review by sdiao — APPROVED** [review]

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs (hunk 7 above)
Comment (peifeng-unity) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs:76]: I double-checked it and they are not the same thing. The `velocity` is the linear velocity in the world space, and the `jointVelocity` is the velocity in the reduced space. The `GetPosition()` and `GetEffort()` methods are both in the reduced space, so I imagined that the `GetVelocity()` should be in the reduced space as well.

**Pull request merged**